pkgs/applications/qubes/qubes-vmm-xen/default.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  fetchpatch,
+  callPackage,
+  ocaml-ng,
+  xen,
+  ...
+}@genericDefinition:
+
+let
+  pname = "qubes-vmm-xen";
+  branch = "4.19";
+  version = "4.19.0";
+  revision = "3";
+  latest = true;
+
+  xenPatches = import ../../virtualization/xen/generic/patches.nix {
```

Comment:
Maybe it'll be better to have it as part of xen pkgset?

Reply:
It is also causes nixpkgs-vet failure.

Reply:
No way around it. I'll need to put Xen in build-support.

Reply:
It is allowed for packages which aren't by-name, I think?

Reply:
yes, but new packages outside by-name aren't allowed.

Reply:
I believe they are allowed, if they can't work with by-name.

And in this case they don't, as they can't receive all the dependencies using by-name.
Maybe it'll be ok if we provide xenPatches in xenPackages pkgset?

Reply:
In #345192 I've moved the generic patch handling (`xenPatches` here, `upstreamPatches` in the normal Xen packages) to the generic builder. https://github.com/NixOS/nixpkgs/blob/787325c0eca67860811cb3a93ea97fb1b64cc54a/pkgs/build-support/xen/default.nix#L116-L124

Reply:
Since that PR doesn't evaluate I have no idea if it works.

Reply:
Ah, you did split it, I haven't noticed that